pkg/plugins/golang/v3/api.go

```diff
@@ -170,23 +172,28 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 	return nil
 }
 
-func (p *createAPISubcommand) PreScaffold(machinery.Filesystem) error {
+func (p *createAPISubcommand) PreScaffold(fs machinery.Filesystem) error {
 	// check if main.go is present in the root directory
 	if _, err := os.Stat(DefaultMainPath); os.IsNotExist(err) {
 		return fmt.Errorf("%s file should present in the root directory", DefaultMainPath)
 	}
 
+	// Check if we use workspaces by presence of go.work, fallback to go.mod otherwise
+	p.useWorkspaces = true
+	if _, err := fs.FS.Stat("go.work"); err != nil {
+		p.useWorkspaces = false
+	}
+
 	return nil
 }
 
 func (p *createAPISubcommand) Scaffold(fs machinery.Filesystem) error {
-	scaffolder := scaffolds.NewAPIScaffolder(p.config, *p.resource, p.force)
+	scaffolder := scaffolds.NewWorkspaceAPIScaffolder(p.config, *p.resource, p.force, p.useWorkspaces)
```

Comment:
I do not think that we should change `NewAPIScaffolder ` , the info about the layout should be in the config so that we need to need to change anything here,